### DISCUSSION TITLE
`Literal["a"]` is allowed as `str` but `list[Literal["a"]]` as `list[str]` is not. Why?

### DISCUSSION THREAD

**Discussion by rafrafek** [microsoft/pyright#6135] [General]
Hello,

Why is this allowed:
```python
from typing import Literal

def demo(foobar: str):
    print(foobar)

def foo(bar: Literal["a"]):
    demo(bar)  # Ok
```

but this is not:
```python
from typing import Literal

def demo(foobar: list[str]):
    print(foobar)

def foo(bar: list[Literal["a"]]):
    demo(bar)  # Error
```

**Comment by erictraut** [comment]
The type parameter for `list` is invariant, so a value of type `list[Literal['a']]` is not type compatible with `list[str]`. For a further explanation, refer to [this documentation](https://microsoft.github.io/pyright/#/type-concepts?id=generic-types).

**Reply by rafrafek** [reply]
Thanks!
